lib/SIL/Utils/FieldSensitivePrunedLiveness.cpp

```diff
@@ -508,18 +508,17 @@ void FieldSensitivePrunedLiveBlocks::updateForUse(
   auto *bb = user->getParent();
   getBlockLiveness(bb, startBitNo, endBitNo, resultingLivenessInfo);
 
-  for (auto pair : llvm::enumerate(resultingLivenessInfo)) {
-    unsigned index = pair.index();
+  for (unsigned index : indices(resultingLivenessInfo)) {
     unsigned specificBitNo = startBitNo + index;
-    switch (pair.value()) {
+    switch (resultingLivenessInfo[index]) {
     case LiveOut:
     case LiveWithin:
       continue;
     case Dead: {
       // This use block has not yet been marked live. Mark it and its
       // predecessor blocks live.
       computeScalarUseBlockLiveness(bb, specificBitNo);
-      resultingLivenessInfo.push_back(getBlockLiveness(bb, specificBitNo));
+      resultingLivenessInfo[index] = getBlockLiveness(bb, specificBitNo);
```

Comment:
Can we add an assert before this loop that `resultingLivenessInfo.size() == endBitNo - startBitNo` or something like that? Just to help verify and document the correspondence here, since we're relying on `getBlockLiveness` to size the array correctly now.